azurerm/internal/services/network/network_interface_nat_rule_association_resource_test.go

```diff
@@ -131,59 +131,48 @@ func (t NetworkInterfaceNATRuleAssociationResource) Exists(ctx context.Context,
 	return utils.Bool(found), nil
 }
 
-func testCheckNetworkInterfaceNATRuleAssociationDisappears(resourceName string) resource.TestCheckFunc {
-	return func(s *terraform.State) error {
-		client := acceptance.AzureProvider.Meta().(*clients.Client).Network.InterfacesClient
-		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
-
-		// Ensure we have enough information in state to look up in API
-		rs, ok := s.RootModule().Resources[resourceName]
-		if !ok {
-			return fmt.Errorf("Not found: %s", resourceName)
-		}
+func (NetworkInterfaceNATRuleAssociationResource) destroy(ctx context.Context, client *clients.Client, state *terraform.InstanceState) error {
+	nicID, err := azure.ParseAzureResourceID(state.Attributes["network_interface_id"])
```

Comment:
As above.

Reply:
this can for the destroy, but can't be for the Exists (since it's not a NIC ID, but a IP Configuration ID)